src/components/multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { FC, useRef, useState } from "react";
+import IconExpand from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/ExpandMore.svg";
+import { useMultipleSelection, useSelect } from "downshift";
+import clsx from "clsx";
+import { useClickAway, useDeepCompareEffect } from "react-use";
+import CheckBox from "../checkbox/Checkbox";
+import {
+  MultiselectExternalUpdate,
+  MultiselectOption
+} from "../../core/utils/types/multiselect-types";
+import useMultiselectOptions from "./useMultiselectOptions";
+import { useText } from "../../core/utils/text";
+
+export type MultiselectProps = {
+  dataCy?: string;
+  caption?: string;
+  options: MultiselectOption[];
+  defaultValue?: MultiselectOption[];
+  updateExternalState?: MultiselectExternalUpdate;
+};
+
+const Multiselect: FC<MultiselectProps> = ({
+  dataCy = "multiselect",
+  caption,
+  options,
+  defaultValue = [],
+  updateExternalState
+}) => {
+  const t = useText();
+  const ref = useRef(null);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const updateState = (
+    updateKey: string | undefined,
+    value: MultiselectOption[]
+  ) => {
+    if (!updateExternalState || !updateKey) return;
+    updateExternalState.externalUpdateFunction({
+      key: updateKey,
+      value
+    });
+  };
+
+  const { allOptions } = useMultiselectOptions(options, {
+    item: "advancedSearchFilterAllText",
```

Comment:
This component is indicating (via the naming and placement in the project) that it is a generic component. But there are still remains of advanced search here and there. I think you need to generalise those as well.